nf_core/pipelines/lint/pipeline_if_empty_null.py

```diff
@@ -0,0 +1,58 @@
+import fnmatch
+import logging
+import os
+import re
+from pathlib import Path
+
+log = logging.getLogger(__name__)
+
+
+def pipeline_if_empty_null(self, root_dir=None):
+    """Check for ifEmpty(null)
+
+    There are two general cases for workflows to use the channel operator `ifEmpty`:
+        1. `ifEmpty( [ ] )` to ensure a process executes, for example when an input file is optional (although this can be replaced by `toList()`).
+        2. When a channel should not be empty and throws an error `ifEmpty { error ... }`, e.g. reading from an empty samplesheet.
+
+    There are multiple examples of workflows that inject null objects into channels using `ifEmpty(null)`, which can cause unhandled null pointer exceptions.
+    This lint test throws warnings for those instances.
+    """
+    passed = []
+    warned = []
+    file_paths = []
+    pattern = re.compile(r"ifEmpty\(\s*null\s*\)")
```

Comment:
Let's also consider cases where people add a space before the parenthesis
```suggestion
    pattern = re.compile(r"ifEmpty\s*\(\s*null\s*\)")
```